factor/sms/classes/local/smsgateway/aws_sns.php

```diff
@@ -27,8 +27,9 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-require_once($CFG->dirroot . '/local/aws/sdk/aws-autoloader.php');
-use Aws\Sns\SnsClient;
+if (file_exists($CFG->dirroot . '/local/aws/version.php')) {
```

Comment:
I dont think the check for the AWS API should be here. You likely shouldnt be able to reach this code path at all if it isnt installed, should probably not even be an option at the settings level.

if we dont load the SDK, it will just exception when we try to use it anyway